File: shared/api-contracts/costs.kt
/**
 * Cost Estimation API Contracts (Kotlin Mirror)
 *
 * Mirror of the TypeScript interfaces in costs.ts.
 * The actual backend data classes live in the backend module.
 *
 * @see costs.ts for the TypeScript source of truth
 */
package contracts

/** Berlin districts (Bezirke) */
enum class Bezirk(val displayName: String) {
    MITTE("Mitte"),
    FRIEDRICHSHAIN_KREUZBERG("Friedrichshain-Kreuzberg"),
    PANKOW("Pankow"),
    CHARLOTTENBURG_WILMERSDORF("Charlottenburg-Wilmersdorf"),
    SPANDAU("Spandau"),
    STEGLITZ_ZEHLENDORF("Steglitz-Zehlendorf"),
    TEMPELHOF_SCHOENEBERG("Tempelhof-Schöneberg"),
    NEUKOELLN("Neukölln"),
    TREPTOW_KOEPENICK("Treptow-Köpenick"),
    MARZAHN_HELLERSDORF("Marzahn-Hellersdorf"),
    LICHTENBERG("Lichtenberg"),
    REINICKENDORF("Reinickendorf")
}

/** Rent range in EUR/month */
data class RentRange(
    val min: Double,
    val max: Double,
    val median: Double
)

/** Monthly cost breakdown for a neighborhood */
data class CostEstimate(
    val bezirk: Bezirk,
    val displayName: String,
    val rooms: Int,
    val rentRange: RentRange,
    val utilities: Double,
    val transport: Double,
    val groceries: Double,
    val totalEstimated: Double
)

/** Neighborhood profile for the explorer view */
data class NeighborhoodProfile(
    val bezirk: Bezirk,
    val displayName: String,
    val vibe: String,
    val commuteMinutes: Int,
    val highlights: List<String>
)

File: shared/api-contracts/costs.ts
/**
 * Cost Estimation API Contracts
 *
 * Interfaces for the cost estimation endpoint that provides
 * living cost data for Berlin neighborhoods (Bezirke).
 *
 * @see costs.kt for the Kotlin mirror
 * @see backend/src/.../controller/CostController.kt for the endpoint
 */

/** Berlin districts (Bezirke) */
export type Bezirk =
  | 'mitte'
  | 'friedrichshain-kreuzberg'
  | 'pankow'
  | 'charlottenburg-wilmersdorf'
  | 'spandau'
  | 'steglitz-zehlendorf'
  | 'tempelhof-schoeneberg'
  | 'neukoelln'
  | 'treptow-koepenick'
  | 'marzahn-hellersdorf'
  | 'lichtenberg'
  | 'reinickendorf';

/** Query parameters for GET /api/v1/costs/estimate */
export interface CostEstimateQuery {
  /** District to estimate costs for */
  bezirk: Bezirk;
  /** Number of rooms in the apartment */
  rooms: number;
}

/** Monthly cost breakdown for a neighborhood */
export interface CostEstimate {
  /** The district this estimate is for */
  bezirk: Bezirk;
  /** Display name of the district */
  displayName: string;
  /** Number of rooms requested */
  rooms: number;
  /** Rent range in EUR/month */
  rentRange: {
    min: number;
    max: number;
    median: number;
  };
  /** Estimated monthly utility costs (Nebenkosten) in EUR */
  utilities: number;
  /** Monthly public transport pass (BVG AB zone) in EUR */
  transport: number;
  /** Estimated monthly grocery costs in EUR */
  groceries: number;
  /** Total estimated monthly costs (using median rent) */
  totalEstimated: number;
}

/** Neighborhood profile for the explorer view */
export interface NeighborhoodProfile {
  /** District identifier */
  bezirk: Bezirk;
  /** Display name */
  displayName: string;
  /** Short character description */
  vibe: string;
  /** Average commute time to city center in minutes */
  commuteMinutes: number;
  /** Key highlights (3-5 bullet points) */
  highlights: string[];
}
